eva_submission/eload_ingestion.py

```diff
@@ -262,23 +264,39 @@ def run_accession_workflow(self):
             raise e
         return output_dir
 
-    def run_variant_load_workflow(self):
+    def run_variant_load_workflow(self, vep_version, vep_cache_version, skip_annotation):
         output_dir = self.create_nextflow_temp_output_directory(base=self.project_dir)
         job_props = variant_load_props_template(
-                project_accession=self.project_accession,
-                # TODO currently there is only ever one of these in the config, even if multiple analyses/files
-                analysis_accession=self.eload_cfg.query('brokering', 'ena', 'ANALYSIS'),
-                aggregation=self.eload_cfg.query(self.config_section, 'aggregation'),
-                study_name=self.get_study_name(),
-                fasta=self.eload_cfg.query('submission', 'assembly_fasta'),
-                output_dir=self.project_dir.joinpath(project_dirs['transformed']),
-                annotation_dir=self.project_dir.joinpath(project_dirs['annotation']),
-                stats_dir=self.project_dir.joinpath(project_dirs['stats']),
-                db_name=self.eload_cfg.query(self.config_section, 'database', 'db_name'),
-                vep_species=self.get_vep_species(),
-                vep_version=self.eload_cfg.query(self.config_section, 'variant_load', 'vep', 'version'),
-                vep_cache_version=self.eload_cfg.query(self.config_section, 'variant_load', 'vep', 'cache_version')
+            project_accession=self.project_accession,
+            # TODO currently there is only ever one of these in the config, even if multiple analyses/files
+            analysis_accession=self.eload_cfg.query('brokering', 'ena', 'ANALYSIS'),
+            aggregation=self.eload_cfg.query(self.config_section, 'aggregation'),
+            study_name=self.get_study_name(),
+            fasta=self.eload_cfg.query('submission', 'assembly_fasta'),
+            output_dir=self.project_dir.joinpath(project_dirs['transformed']),
+            annotation_dir=self.project_dir.joinpath(project_dirs['annotation']),
+            stats_dir=self.project_dir.joinpath(project_dirs['stats']),
+            db_name=self.eload_cfg.query(self.config_section, 'database', 'db_name'),
+            vep_species=self.get_vep_species(),
+            vep_version=vep_version,
+            vep_cache_version=vep_cache_version,
+            annotation_skip=skip_annotation
         )
+        if skip_annotation is False and vep_version is None:
+            coll_name = job_props['db.collections.annotations.name']
```

Comment:
I'm wondering if we can pull this collection name from a common source (e.g. [here](https://github.com/EBIvariation/configuration/blob/0d1f9f3e6f41c7c05bb7f9cf51fc4905acc41fb5/eva-maven-settings.xml#L121)) in both the variant load properties template and the getting vep versions method, rather than having one getting it from the other?

Then we can have vep versions/skipping figured out and set in the eload config before we create the properties file, and maybe before even entering the `run_variant_load_workflow` method (which is starting to be responsible for a lot more than just running a workflow).

(If this approach is too much work to get right for this PR then that's ok too, it looks correct to me and we can always refactor later.)

Reply:
will refactor it in the next PR.